pkgs/applications/window-managers/tinywl/default.nix

```diff
@@ -4,25 +4,29 @@
 
 stdenv.mkDerivation {
   pname = "tinywl";
-  inherit (wlroots) version src;
-
-  sourceRoot = "${wlroots.src.name}/tinywl";
+  inherit (wlroots) version src patches postPatch;
 
   nativeBuildInputs = [ pkg-config wayland-scanner ];
   buildInputs = [ libxkbcommon pixman udev wayland wayland-protocols wlroots ];
 
+  buildPhase = ''
+    runHook preBuild
+    make -C tinywl
+    runHook postBuild
+  '';
```

Comment:
This could be `makeFlags = [ "-C" "tinywl" ];`, and then there'd be no need for a custom build phase.